src/Orleans/AzureUtils/StatsTableDataManager.cs

```diff
@@ -136,20 +137,31 @@ public Task ReportStats(List<ICounter> statsCounters)
         private StatsTableData PopulateStatsTableDataEntry(ICounter statsCounter)
         {
             string statValue = statsCounter.IsValueDelta ? statsCounter.GetDeltaString() : statsCounter.GetValueString();
-            var entry = new StatsTableData { StatValue = statValue };
-
-            if ("0".Equals(entry.StatValue))
+            if ("0".Equals(statValue))
             {
                 // Skip writing empty records
                 return null;
             }
 
             counter++;
-            var time = DateTime.UtcNow;
-            var timeStr = time.ToString(DATE_FORMAT, CultureInfo.InvariantCulture);
+            var entry = new StatsTableData { StatValue = statValue };
+            
+            // We store the statistics grouped by an hour in the same partition and sorted by reverse ticks within the partition.
+            // Since by default Azure table stores Entities in ascending order based on the Row Key - if we just used the current ticks
+            // it would return the oldest entry first.
+            // If we store the rows in the reverse order (some max value - current ticks), it will return the latest most recent entry first.
+            // More details here:
+            // http://gauravmantri.com/2012/02/17/effective-way-of-fetching-diagnostics-data-from-windows-azure-diagnostics-table-hint-use-partitionkey/
+            // https://alexandrebrisebois.wordpress.com/2014/06/16/using-time-based-partition-keys-in-azure-table-storage/
+            // https://stackoverflow.com/questions/1004698/how-to-truncate-milliseconds-off-of-a-net-datetime
+            
+            // Our format:
+            // PartitionKey:  DeploymentId$ReverseTimestampToTheNearestHour 
+            // RowKey:  ReverseTimestampToTheNearestSecond$Name$counter 
 
+            var now = DateTime.UtcNow;
             // number of ticks remaining until the year 9683
-            var ticks = DateTime.MaxValue.Ticks - time.Ticks;
+            var ticks = DateTime.MaxValue.Ticks - now.Ticks;
 
             // partition the table according to the deployment id and hour
             entry.PartitionKey = string.Join("$", deploymentId, string.Format("{0:d19}", ticks - ticks % TimeSpan.TicksPerHour));
```

Comment:
That was Richard's original PR. I did not touch it. After reading all the blogs I got convinced he implemented what was described in the blogs and that we want it. Am I wrong?